x/evm/types/command.go

```diff
@@ -138,6 +138,25 @@ func NewApproveContractCallCommand(
 	}
 }
 
+// NewApproveContractCallCommandFromGeneralMessage creates a command to approve contract call
+func NewApproveContractCallCommandFromGeneralMessage(
+	chainID sdk.Int,
+	keyID multisig.KeyID,
+	msg nexus.GeneralMessage,
+) Command {
+	contractAddress := common.HexToAddress(msg.Receiver)
+	dummyTxID := common.BytesToHash(make([]byte, common.HashLength))
+	payloadHash := common.BytesToHash(msg.PayloadHash)
+	commandID := NewCommandID([]byte(fmt.Sprintf("%s_%s_%s", contractAddress, payloadHash, msg.ID)), chainID)
```

Comment:
Why is `msg.ID` not enough?

Reply:
yep, specifically, `msg.ID.ID` since it's created from the tx hash and nonce

Reply:
`msg.ID.ID` will be enough, this is leftover from before I was using the nonce